bycycle/cyclepoints.py

```diff
@@ -257,26 +227,25 @@ def extrema_interpolated_phase(sig, ps, ts, zerox_rise=None, zerox_decay=None):
     pha_tnpi = np.zeros(sig_len) * np.nan
 
     # If specified, assign phases to zero-crossings
-    if zerox_rise is not None:
-        pha_tpi[zerox_rise] = -np.pi / 2
-        pha_tnpi[zerox_rise] = -np.pi / 2
-    if zerox_decay is not None:
-        pha_tpi[zerox_decay] = np.pi / 2
-        pha_tnpi[zerox_decay] = np.pi / 2
+    if rises is not None:
+        pha_tpi[rises] = -np.pi / 2
+        pha_tnpi[rises] = -np.pi / 2
+    if decays is not None:
+        pha_tpi[decays] = np.pi / 2
+        pha_tnpi[decays] = np.pi / 2
 
     # Define phases
-    pha_tpi[ps] = 0
-    pha_tpi[ts] = np.pi
-    pha_tnpi[ps] = 0
-    pha_tnpi[ts] = -np.pi
+    pha_tpi[peaks] = 0
+    pha_tpi[troughs] = np.pi
+    pha_tnpi[peaks] = 0
+    pha_tnpi[troughs] = -np.pi
 
     # Interpolate to find all phases
     pha_tpi = np.interp(times, times[~np.isnan(pha_tpi)], pha_tpi[~np.isnan(pha_tpi)])
     pha_tnpi = np.interp(times, times[~np.isnan(pha_tnpi)], pha_tnpi[~np.isnan(pha_tnpi)])
 
-    # For the phase time series in which the trough is negative pi:
-    # Replace the decaying periods with these periods in the phase time
-    # series in which the trough is pi
+    # For the phase time series in which the trough is negative pi, replace the decaying
+    #   periods with these periods in the phase time series in which the trough is pi
     diffs = np.diff(pha_tnpi)
     diffs = np.append(diffs, 99)
```

Comment:
this `99` is a weird magic number for quite unclear reasons. I think we should figure out what this is / make a note.

Reply:
99 is added to keep the length of the interpolated phase array == length of the signal. The negative pi phase array is always used for the last sample. I replaced 99 with np.nan and left a comment (3c42063).